docs/pages/hdmi_cec_source_tests.md

```diff
@@ -0,0 +1,176 @@
+| 3| [CEC Synchronous Transmission](#cec-synchronous-transmission)| Transmitting CEC frames and acknowledging them |
+| 4| [CEC Receive functionality](#cec-receive-functionality)| Receiving CEC information from other devices and communicating it to the above layers through registered callback functions |
+| 5| [CEC HotPlug Functionality](#cec-hotplug-functionality)| Managing CEC during Hotplug and HotUnplug events |
+
+-----------
+
+## Logical Address Discovery
+
+|S.No.|Test Functionality|Description|HAL APIs|L2|L3|Control plane requirements|
+|-----|------------------|-----------|--------|--|--|--------------------------|
+| 1 |[Logical address](#logical-address-discovery)|Get the logical address of the `DUT`. This will add the logical address, as per source functionality. |HdmiCecGetLogicalAddress|Y|NA
+
+### Emulator Requirements
+
+- Boot with control configuration with various configurations having a predefined set of nodes:
+  - configuration to support the discovery of logical addresses. The caller provides the logical address, and HAL checks the availability of this address and feedback the same to the caller. 
+  - Verify for the valid logical address and return the appropriate error code based on the logical address availability.  
+
+### Control Plane Requirements
+
+- The control plane will allow removing or adding a node to the network.
+  - allowing adding/removing node sink node logical address 0
+  - allowing adding/removing source node
+  - Support the CEC commands from the external devices on L3 Test Cases. 
+
+### Physical Address
+
+|S.No.|Test Functionality|Description|HAL APIs|L2|L3|Control plane requirements|
+|-----|------------------|-----------|--------|--|--|--------------------------|
+| 1| [Physical Address](#physical-address)| Verify the valid physical address allocated through the HAL function.|HdmiCecGetPhysicalAddress| Y  |NA||
+| | | Verify the physical addresses allocated by connecting a source and sink device through an HDMI switch.|NA | NA | Y  | Enable the television connected to my Tele first so that it can declare its physical address first.  |
+
+#### Emulator Requirements - Physical Address
+
+- Boot control configuration to setup the cec network nodes
+  - A source and sink devices are connected to the network
+
+#### Control Plane Requirements - Physical Address
+
+- Control plane will allow removing or adding a node to the network.
+  - allowing adding/removing node sink node
+  - allowing adding/removing node source node
+
+### CEC Synchronous Transmission
+
+|S.No.|Test Functionality|Description|HAL APIs|L2|L3|Control plane requirements|
+|-----|------------------|-----------|--------|--|--|--------------------------|
+| 3| [CEC Transmission](#cec-transmission)| Verify the correct transmission of the supported CEC commands (as per 1.4b HDMI CEC spec) to the connected device and ensure it is acknowledged properly.| HdmiCecTx | NA | Y  ||
```

Comment:
Can we check by sending the commands to a sink device and also another source device connected on the network.  Basically this test case can be made into two.

Reply:
It would need to be a sink device, I think given that the sink device would 'discover' the valid logical address of the connected sink device, I don't think it can do that too another source. Atleast that would be outside of expected working parameters